Failure to build debug version under Windows

After the latest change (always link statically under Windows), debug builds of kiwisolver (and possibly other modules) fail with linker errors due to missing debug symbols:
```
variable.obj : error LNK2001: unresolved external symbol _invalid_parameter
libcpmtd.lib(locale.obj) : error LNK2001: unresolved external symbol _invalid_parameter
libcpmtd.lib(wlocale.obj) : error LNK2001: unresolved external symbol _invalid_parameter
libcpmtd.lib(xlocale.obj) : error LNK2001: unresolved external symbol _invalid_parameter
constraint.obj : error LNK2001: unresolved external symbol _invalid_parameter
expression.obj : error LNK2001: unresolved external symbol _invalid_parameter
solver.obj : error LNK2001: unresolved external symbol _invalid_parameter
term.obj : error LNK2001: unresolved external symbol _invalid_parameter
libcpmtd.lib(_tolower.obj) : error LNK2001: unresolved external symbol _calloc_dbg
libcpmtd.lib(locale.obj) : error LNK2001: unresolved external symbol _calloc_dbg
libcpmtd.lib(wlocale.obj) : error LNK2001: unresolved external symbol _calloc_dbg
libcpmtd.lib(xlocale.obj) : error LNK2001: unresolved external symbol _calloc_dbg
...
```

The reason can be seen in the warning:
```
cl : Command line warning D9025 : overriding '/MDd' with '/MT'
```
e.g. while overriding the flag the debug mode is ignored.

--- a/cppy/__init__.py
+++ b/cppy/__init__.py
@@ -52,13 +52,21 @@ class CppyBuildExt(build_ext):
                 # linking for `VCRUNTIME140.dll`, `VCRUNTIME140_1.dll`, and the UCRT.
                 # This avoids requiring specific versions of `MSVCP140.dll`, while
                 # keeping shared state with the rest of the Python process/extensions.
-                ext.extra_compile_args.append("/MT")
+                is_debug = hasattr(sys, "gettotalrefcount")  # only present in Python debug build
+
+                # Mixing debug and release code is bad practice under Windows. The problem is that the
+                # different versions can depend on different fundamental parts of the C++ runtime library,
+                # such as how memory is allocated, structures for things like iterators might be
+                # different, extra code could be generated to perform operations (e.g. checked iterators).
+                # As a consequence we build as debug if python is built with debug symbols.
+                debug_ext = "d" if is_debug else ""
+                ext.extra_compile_args.append(f"/MT{debug_ext}")
                 ext.extra_link_args.extend(
                     [
-                        "ucrt.lib",
-                        "vcruntime.lib",
-                        "/nodefaultlib:libucrt.lib",
-                        "/nodefaultlib:libvcruntime.lib",
+                        f"ucrt{debug_ext}.lib",
+                        f"vcruntime{debug_ext}.lib",
+                        f"/nodefaultlib:libucrt{debug_ext}.lib",
+                        f"/nodefaultlib:libvcruntime{debug_ext}.lib",
                     ]
                 )
                 if os.environ.get("CPPY_DISABLE_FH4"):

--- a/releasenotes.rst
+++ b/releasenotes.rst
@@ -1,9 +1,13 @@
 Cppy Release Notes
 ==================
 
+1.3.1 - unreleased
+-------------------
+- on Windows link to debug runtime lib when building with a debug Python PR #29
+
 1.3.0 - 26/11/2024
 ------------------
-- prevent liinking dynamically to MSVCP on windows
+- prevent linking dynamically to MSVCP on windows
 
 1.2.1 - 03/30/2022
 ------------------

